uaclient/contract.py

```diff
@@ -212,13 +266,28 @@ def _request_machine_token_update(
         response, headers = self.request_url(url, **kwargs)
         if headers.get("expires"):
             response["expires"] = headers["expires"]
-        if not detach:
-            self.cfg.write_cache("machine-token", response)
-            util.get_machine_id.cache_clear()
-            machine_id = response.get("machineTokenInfo", {}).get(
-                "machineId", data.get("machineId")
-            )
-            self.cfg.write_cache("machine-id", machine_id)
+        return response
+
+    def _get_updated_contract_info(
```

Comment:
Does this need to be a separate function? Can this all just go in `get_updated_contract_info`?

Reply:
Yeah, makes sense. We can reduce the amount of code diff.